src/scheduler.rs

```diff
@@ -333,7 +341,23 @@ async fn smtp_loop(ctx: Context, started: Sender<()>, smtp_handlers: SmtpConnect
                         Some(err) => connection.connectivity.set_err(&ctx, err).await,
                     }
 
-                    interrupt_info = idle_interrupt_receiver.recv().await.unwrap_or_default();
+                    if let Some(timeout) = timeout {
```

Comment:
Mabye but a "big picture" comment here (or somewhere else) along the lines `If send_smtp_messages() failed, we set a time-out for the fake-idle so that sending is retried (at the latest) after the timeout. If sending fails again, we increase the timeout exponentially, in order not to do lots of unnecessary retries.`